src/Microsoft.AspNetCore.Razor.Language/Extensions/SectionDirectivePass.cs

```diff
@@ -37,7 +37,9 @@ protected override void ExecuteCore(RazorCodeDocument codeDocument, DocumentInte
 
                 for (; i < directive.Node.Children.Count; i++)
                 {
+                    // Move non-token children over to the section node so we don't have double references to children nodes.
                     section.Children.Add(directive.Node.Children[i]);
+                    directive.Node.Children.RemoveAt(i--);
```

Comment:
It would be better to change this to a while loop. This is just.... an abomination.